wit/couchbase/fts.wit

```diff
@@ -0,0 +1,216 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Support Full Text Search (FTS) provided by Couchbase, which spans:
+///   - Vectors
+///   - Structured/Unstructured text
+///   - Dates
+///   - Numbers
+///   - CIDR notation
+///   - Geospatial data
+///
+/// Reference: https://docs.couchbase.com/server/current/search/search.html
+interface fts {
+  use types.{
+    request-span, document, document-field-name, collection-name, sort-direction, 
+    bucket-error, search-index-name, retry-strategy
+    };
+
+  variant fts-bucket-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Buckets for use with FTS
+  ///
+  /// NOTE: This is unnecessary (and may be implemented as a no-op) for Couchbase versions greater than 6.5
+  resource fts-bucket {
+    constructor(name: string);
+    /// Wait until a bucket is ready to use
+    ///
+    /// This should be done on couchbase versions earlier than 6.5, before performing any search queries.
+    wait-until-ready: func() -> result<_, bucket-error>;
```

Comment:
In the context of WASMCloud, which will probably handle the consistency of the _infrastructure_ before allowing the usage of its business functions, I think it would be beneficial to provide a `wait-mutations`, so the bootstrap process knows when its finished.
A search can set the consistency to `RequestPlus` but this means the wasm business functions are already made available, so this is not optimal.

Reply:
This was added mostly to enable [this part of the docs](https://docs.couchbase.com/go-sdk/current/howtos/full-text-searching-with-sdk.html#getting-started) -- am I understanding right that search should *also* expose/perform a wait on mutations beforehand? 

Note that it's actually up to the provider what `wait-until-ready` means -- it can include other kinds of waits, and it's also possible that the caller doesn't use it at all (and the provider has to handle that case or throw an error/etc)

Maybe this should actually be in the domain of the provider to properly implement, and the related bucket should be passed to the search query? Is there an automatable way to tell the connection between a search index and the bucket it depends on/that has to be ready before use?

Reply:
Hey @JesusTheHun what do you think about this one? I think I've addressed everything else, but I only see waits for ready on cluster/buckets in the Go SDK

Reply:
Sorry I misunderstood the context. It was in the fts.wit file so I thought it was about search query indexes, but it's not. You can ignore my comment.